added debian-related instructions and fixed some arch-only lines

POMATu opened the pull request:
after my success with making suspend work on debian bookworm same way it works on arch I have added debian-related instructions and fixed some of the confusing arch-only lines 
I think its important to include my .config for compiling current kernel on debian because i think this is the main reason why suspend works for me but doesnt works on any of the current kernels from repos
you might remove it later when there will be suspend-enabled packages for debian-type distros in the repos

the kernel i reference is:
Linux macbookpro 6.9.7 #1 SMP PREEMPT_DYNAMIC Sun Jun 30 09:07:34 EDT 2024 x86_64 GNU/Linux

AdityaGarg8 commented on docs/guides/kernel.md:
````diff
@@ -41,8 +41,19 @@ done
+```bash
+sudo cp /boot/config-$(uname -r) .config
+```
+**for Debian-based distros:** or you might as well just [download the debian config](../attachments/debian.config) that was observed to have [suspend-workaround](../guides/postinstall.md) working on debian bookworm. Name the file as .config and place in current directory.
````
Kernel configs change over time. You cannot make people use a single kernel config forever. Instead you need to find out what's special in this config that makes things work.

POMATu replied:
removed in recent commit

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -18,18 +18,19 @@ Installing a kernel with support for T2 Macs is required in order to get the Key
+| Ubuntu based distros                  | <https://github.com/t2linux/T2-Debian-and-Ubuntu-Kernel> |
+| Debian based distros                  | <https://github.com/t2linux/T2-Debian-and-Ubuntu-Kernel> |
+| Debian based distros                  | <https://github.com/andersfugmann/T2-Debian-Kernel> |
+| Debian based distros (suspend config) | <https://wiki.t2linux.org/guides/kernel/> |
```
Not the right place to keep this.

POMATu replied:
removed

AdityaGarg8 commented on docs/guides/postinstall.md:
````diff
@@ -172,9 +173,73 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+This script was written specifically for Debian Bookworm tiny-dfr touchbar issue, but you will loose ability to control keyboard brightness with touchbar. For some reason unloading ```hid_appletb_kbd``` helps to maintain touchbar in alive state but then this module trips and doesnt work anymore.
+You can still control keyboard brightness with just bash tho:
+```bash
+echo 8192 > /sys/class/leds/\:white\:kbd_backlight/brightness
````
This command is specific to only MacBookPro16,x and MacBookAir9,1

AdityaGarg8 replied:
Also use sudo tee

POMATu replied:
fixed

AdityaGarg8 reviewed (changes requested):
Spacing of the scripts is not proper. Your lint will fail if I approve this.

AdityaGarg8 commented on docs/guides/kernel.md:
```diff
@@ -23,8 +23,10 @@ If your distro is not one of the distros with documentation on this Wiki, you ma
 mkdir build && cd build
-git clone --depth=1 https://github.com/t2linux/linux-t2-patches patches
-
+git clone https://github.com/t2linux/linux-t2-patches patches
```
Remove this change. It's a temporary thing and will automatically get fixed when 6.10 gets released

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -142,6 +142,24 @@ blacklist cdc_mbim" >> /etc/modprobe.d/blacklist.conf'
 
 Please note that this internal ethernet interface is required for various services including touchid that there currently is no Linux support for. In the future, if any of these services are supported, you'll need to undo this.
 
+If this didn't work for you, you can also tell NetworkManager directly to ignore specific interfaces
```
We have a udev rule as a better option for this. Will add after this PR. So remove this.

AdityaGarg8 commented on docs/guides/postinstall.md:
````diff
@@ -172,9 +190,105 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+     ```
+and fix the pathes in service script if they differ for your system
+
+4. If you having problems with touchbar being dead after restoring state from suspend then you might as well try the following version of the script:
````
Instead of writing the almost same thing again, add instructions how to modify the previous service file

sharpenedblade replied:
Its better to put this in a user systemd service in /etc.

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +190,105 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
-    To check, run: `zcat /proc/config.gz | grep "CONFIG_MODULE_FORCE_UNLOAD"`
+    This seems to be working with `CONFIG_MODULE_FORCE_UNLOAD=y` in the kernel config when kernel is compiled with [kernel compilation instructions](https://wiki.t2linux.org/guides/kernel/).
+    
+To check, run: 
```
Again bad spacing. See how notes are written. https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

AdityaGarg8 replied:
You haven't fixed the spacing yet.

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +190,105 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+
+Without this config option you wont be able to unload the required modules, they will be busy.
+
+6. If touchbar occasionally does not work on boot but works after suspend+restore then you can place this workaround somewhere late after boot
```
Won't it be better to just keep this and remove the extra suspend.service you made here https://github.com/t2linux/wiki/pull/556/files#diff-a9bb6c3fdd453b5d31bf4ce51c2122abccd3d3f1d21d0c2c78f5e783a7bc1b87R212

AdityaGarg8 replied:
Again, this hasn't been addressed yet

POMATu commented:
ok i removed touchbar-poke.service since it failed for me at least once and simplified the footer of the file. Maybe worth including instructions how to bind that script to power button to solve any kind of occasional kernel module reloading issues.

sharpenedblade commented:
> ok i removed touchbar-poke.service since it failed for me at least once and simplified the footer of the file. Maybe worth including instructions how to bind that script to power button to solve any kind of occasional kernel module reloading issues: whether its restoring from suspend or tiny-dfr being dead.

`tiny-dfr` can be fixed in less hacky ways. Its better to get tiny-dfr working by itself, the code is pretty simple so its doable.

AdityaGarg8 commented on docs/guides/kernel.md:
```diff
@@ -23,8 +23,7 @@ If your distro is not one of the distros with documentation on this Wiki, you ma
 mkdir build && cd build
-git clone --depth=1 https://github.com/t2linux/linux-t2-patches patches
-
+git clone https://github.com/t2linux/linux-t2-patches patches
```
Keep --depth 1

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +174,44 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+and fix the pathes in service script if they differ for your system. If you having problems with touchbar being dead after restoring state from suspend then uncomment ExecStartPre line.
 
-3. Enable the service by running: `sudo systemctl enable --now suspend-fix-t2.service`
+5. Enable the service by running: `sudo systemctl enable --now suspend-fix-t2.service`
```
Is this number 4?

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +174,44 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
-    This seems to be working only on Arch with `CONFIG_MODULE_FORCE_UNLOAD=y` in the kernel config.
-    To check, run: `zcat /proc/config.gz | grep "CONFIG_MODULE_FORCE_UNLOAD"`
+
+    This seems to be working with `CONFIG_MODULE_FORCE_UNLOAD=y` in the kernel config when kernel is compiled with [kernel compilation instructions](https://wiki.t2linux.org/guides/kernel/).
```
Why specifically kernel compilation instructions? There are people who know how to compile a kernel better than this guide.

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +174,44 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
 !!! note
-    This seems to be working only on Arch with `CONFIG_MODULE_FORCE_UNLOAD=y` in the kernel config.
-    To check, run: `zcat /proc/config.gz | grep "CONFIG_MODULE_FORCE_UNLOAD"`
+
```
No blank line here

AdityaGarg8 replied:
This is not resolved

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +174,40 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+
+Without this config option you wont be able to unload the required modules, they will be busy.
+
+5. If touchbar occasionally does not work on boot but works after suspend+restore then you can place this workaround somewhere late after boot
```
Why brcmfmac is touchbar? Both have no relation to each other.

AdityaGarg8 commented on docs/guides/postinstall.md:
````diff
@@ -172,9 +174,40 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+     ```bash
+     modprobe -r hid_appletb_kbd
+     modprobe -r brcmfmac_wcc
+     modprobe -r brcmfmac
````
Again, why apple-bce is being removed?

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -172,9 +174,40 @@ S3 suspend has been broken since macOS Sonoma, it has never been fixed, but this
+     modprobe brcmfmac
+     modprobe brcmfmac_wcc
+     
+     touchbar --restart
```
touchbar --restart is an Ubuntu only command. Make it disable and enable tiny-dfr.service instead.

AdityaGarg8 commented on docs/guides/postinstall.md:
```diff
@@ -142,6 +142,7 @@ blacklist cdc_mbim" >> /etc/modprobe.d/blacklist.conf'
 
 Please note that this internal ethernet interface is required for various services including touchid that there currently is no Linux support for. In the future, if any of these services are supported, you'll need to undo this.
 
+
```
This extra line is unnecessary

AdityaGarg8 commented on docs/guides/kernel.md:
```diff
@@ -39,10 +38,24 @@ done
 ### Setting kernel configuration
 
 !!! Info "Using config from lower kernel versions"
+
```
Why extra line here? Remove this.

AdityaGarg8 commented:
Extra line as per https://squidfunk.github.io/mkdocs-material/reference/admonitions/ after `!!!note` is not used in the wiki

AdityaGarg8 commented:
> apparently working with people is not my thing, i am afraid i got bored long time ago and already missed the point what I wanted to contribute with this pull request I got my suspend working on my debian I am grateful for that, maybe I can explain to someone in #debian discord if they ask, but contributing thru exam-type moderation i am not interested, closing

No worries. Thanks for your efforts.
